mountpoint-s3-client/src/s3_crt_client/head_bucket.rs

```diff
@@ -10,6 +10,9 @@ pub enum HeadBucketError {
     #[error("Wrong region: should be {0}")]
     IncorrectRegion(String),
 
+    #[error("The bucket did not exist")]
```

Comment:
Can we include the bucket name in error?

Reply:
probably not necessary since the caller knows what bucket name they passed in

Reply:
It's good thinking. I was on the fence with this one.

>probably not necessary since the caller knows what bucket name they passed in

Yeah, I agree. Besides, I gave it a quick go and it wasn't trivial to get the bucket name in. (Lifetime nonsense.)